lib/petpvcRLPVCImageFilter.txx

```diff
@@ -79,50 +79,95 @@ void RichardsonLucyPVCImageFilter< TInputImage >
     typename TInputImage::Pointer imageEstimate;
     //Set image estimate to the original non-negative PET data for the first iteration.
     imageEstimate = duplicator->GetOutput();
+    imageEstimate->DisconnectPipeline();
+
+    // Create a blank image with same dimensions as input to use for division
+    duplicator->SetInputImage(imageEstimate);
+    duplicator->Update();
+
+    typedef itk::CastImageFilter< TInputImage, TInputImage > CastImageFilterType;
+    typename CastImageFilterType::Pointer castFilter = CastImageFilterType::New();
+    castFilter->SetInput( duplicator->GetOutput() );
+    castFilter->Update();
+    typename TInputImage::Pointer blankImage = castFilter->GetOutput();
+
+    typedef typename TInputImage::PixelType PixelType;
+    blankImage->FillBuffer( itk::NumericTraits< PixelType >::Zero );
+
+    // Image to store result of division
+    typename TInputImage::Pointer dividedImage;
 
     int nMaxNumOfIters =  this->m_nIterations;
     int n=1;
 
     bool bStopped = false;
-
 	
     while ( ( n <= nMaxNumOfIters ) && ( !bStopped ) ) {
-         float fLog = 0.0;   
+         float fLog = 0.0;  
+            // f_k * h
             blurFilter->SetInput( imageEstimate );
-			divideFilter->SetInput1( thresholdFilter->GetOutput() );
-			divideFilter->SetInput2( blurFilter->GetOutput() );
+            blurFilter->Update();
+
+            // Perform f(x) / [ f_k(x) * h ] voxel-by-voxel as itk::DivideFilterType sets image
+            // to max if denominator is 0.
+            ConstImageIterator numeratorIt( thresholdFilter->GetOutput(),  thresholdFilter->GetOutput()->GetLargestPossibleRegion() );
+            ConstImageIterator denomIt( blurFilter->GetOutput(),  blurFilter->GetOutput()->GetLargestPossibleRegion() );
+
+            duplicator->SetInputImage(blankImage);
+            duplicator->Update();
+            dividedImage = duplicator->GetOutput();
+            dividedImage->DisconnectPipeline();
+
+            ImageRegionIterator<TInputImage> divIt( dividedImage, dividedImage->GetLargestPossibleRegion() );
+
+            numeratorIt.GoToBegin();
+            denomIt.GoToBegin();
+            divIt.GoToBegin();
+
+            while ( !divIt.IsAtEnd() )
+    	    {
+				if (denomIt.Get() > 1e-4)
```

Comment:
This is very dangerous as it assumes a certain scale in the image. The STIR solution  is to use a number relative to the max (a median would be better but much more expensive). Even that is creating trouble for parametric images (as it uses a global max), see https://github.com/UCL/STIR/issues/906. However, maybe that isn't so much of a problem here)

Reply:
This is the ratio of the image to the smoothed version of itself, so they should always be scaled similarly (unless the FWHM was huge). I can put the max or median if if you think it will cause a problem though.